Downloader.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  downloadModel.swift
+//  react-native-spokestack
+//
+//  Created by Timothy Willison on 12/7/20.
+//
+
+import Foundation
+import Network
+
+@objc
+public class Downloader: NSObject {
+    private var allowCellular: Bool = false
+    private var refreshModels: Bool = false
+    private let monitor = NWPathMonitor()
+    private var path: NWPath?
+    private var downloadQueue: [URL:(Error?, String?) -> Void] = [:]
+    lazy var pathUpdateHandler = { (path: NWPath) in
+        self.path = path
+        if self.isNetworkAvailable() && !self.downloadQueue.isEmpty {
+            print("Downloader: executing from pathUpdateHandler")
+            for (url, complete) in self.downloadQueue {
+                self.downloadModel(url, complete)
+            }
+            // Reset queue
+            self.downloadQueue = [:]
+        }
+    }
+
+    init(allowCellular: Bool, refreshModels: Bool) {
+        super.init()
+        self.allowCellular = allowCellular
+        self.refreshModels = refreshModels
+        monitor.pathUpdateHandler = pathUpdateHandler
+        // Start monitor on background queue
+        monitor.start(queue: DispatchQueue.global(qos: .default))
+    }
+    
+    deinit {
+        monitor.cancel()
+    }
+    
+    func hasStatus() -> Bool {
+        return self.path != nil
+    }
+
+    func isNetworkAvailable() -> Bool {
+        if let path = self.path {
+            return path.status == .satisfied && (allowCellular || !path.isExpensive)
+        }
+        return false
+    }
+
+    func downloadModel(_ url: URL, _ complete: @escaping (Error?, String?) -> Void) {
+        // Wait until network status is availalbe to attempt downloading
+        if hasStatus() {
```

Comment:
simple check, better to leave the logic inline at the point of use.
```suggestion
        if self.path != nil {
```